pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -352,3 +379,19 @@ func isMaster(machine mapiv1.Machine, client client.Client) bool {
 	}
 	return false
 }
+
+func getClusterInfrastructure(c client.Client) (*configv1.Infrastructure, error) {
+	infra := &configv1.Infrastructure{}
+	key := types.NamespacedName{
+		Namespace: metav1.NamespaceNone,
+		Name:      "cluster",
+	}
+	if err := c.Get(context.TODO(), key, infra); err != nil {
+		return nil, err
+	}
+	return infra, nil
+}
+
+func isBareMetalProvider(platform configv1.PlatformType) bool {
```

Comment:
can we have a unit for this?

Reply:
it tested as part of all reconcile and remediation tests, so I believe it will be duplication, but if you insist I can create small unittest for it